pkg/executables/cmk.go

```diff
@@ -26,17 +27,26 @@ type Cmk struct {
 	writer     filewriter.FileWriter
 	executable Executable
 	config     CmkExecConfig
+	zones []cmkZone
+	domain cmkDomain
+	account cmkAccount
+
 }
 
-// TODO: Add support for domain, account filtering
-func (c *Cmk) ValidateTemplatePresent(ctx context.Context, domain string, zone v1alpha1.CloudStackResourceRef, account string, template v1alpha1.CloudStackResourceRef) error {
-	var filterArg string
+func (c *Cmk) ValidateTemplatePresent(ctx context.Context, domainId string, zoneId string, account string, template v1alpha1.CloudStackResourceRef) error {
+	filterArgs := []string{"list", "templates", "templatefilter=all", "listall=true"}
```

Comment:
This gives the impression that `list` and `templates` are filter args too. Should we have `commandArgs` and `filterArgs` as separate variables?

Reply:
separated into commandArgs and filterArgs.